ai_services/anomaly_detection/data_preprocessing_examples/oci_data_flow_based_examples/example_code/ad_utils.py

```diff
@@ -0,0 +1,116 @@
+import json
+from argparse import ArgumentParser
+
+from oci.ai_anomaly_detection import AnomalyDetectionClient
+from oci.ai_anomaly_detection.models import CreateModelDetails, ModelTrainingDetails, CreateDataAssetDetails, \
+    DataSourceDetailsObjectStorage
+from oci.object_storage import ObjectStorageClient
+from pyspark.sql import SparkSession
+import oci
+import os
+
+DEFAULT_LOCATION = os.path.join('~', '.oci', 'config')
+DEFAULT_PROFILE = "DEFAULT"
+DEFAULT_TARGET_FAP = 0.01
+DEFAULT_TRAINING_FRACTION = 0.7
+
+
+def get_spark_context():
+    return SparkSession.builder.appName("AnomalyDetectionClient").getOrCreate()
+
+
+def get_token_path():
```

Comment:
nit: These are duplicated. I would move them to a separate module rather than copy/paste

Reply:
Thanks for sharing this. This was already handled and so closing this conversation directly.